bpf/bpf_lxc.c

```diff
@@ -173,24 +173,36 @@ static __always_inline int ipv6_l3_from_lxc(struct __ctx_buff *ctx,
 			   orig_dip.p4, *dstID);
 	}
 
+	/* When an endpoint connects to itself via service clusterIP, we need
+	 * to skip the policy enforcement. If we didn't, the user would have to
+	 * define policy rules to allow pods to talk to themselves. We still
+	 * want to execute the conntrack logic so that replies can be correctly
+	 * matched.
+	 */
+	if (hairpin_flow)
+		goto skip_policy_enforcement;
+
 	/* If the packet is in the establishing direction and it's destined
 	 * within the cluster, it must match policy or be dropped. If it's
 	 * bound for the host/outside, perform the CIDR policy check.
 	 */
 	verdict = policy_can_egress6(ctx, tuple, SECLABEL, *dstID,
 				     &policy_match_type, &audited);
+
```

Comment:
nit: extra whitespace

Reply:
Also on purpose, to match the IPv4 code. Alternatively I could remove the space everywhere I guess :-)

Reply:
This is fine.